### PR TITLE
Feat: optimize the definition controllers

### PR DESCRIPTION
**Pull request by barnettZQG**
Signed-off-by: barnettZQG <barnett.zqg@gmail.com>

### Description of your changes

1. Remove `RefreshPackageDiscover`, this is an unuseful code.
2. Merge similar code.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/controller/core.oam.dev/v1alpha2/core/components/componentdefinition/componentdefinition_controller.go [hunk 1]
```diff
@@ -76,59 +73,23 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
-	if !r.matchControllerRequirement(&componentDefinition) {
-		klog.InfoS("skip componentDefinition: not match the controller requirement of componentDefinition", "componentDefinition", klog.KObj(&componentDefinition))
+	if !coredef.MatchControllerRequirement(&componentDefinition, r.controllerVersion, r.ignoreDefNoCtrlReq) {
+		klog.InfoS("skip definition: not match the controller requirement of definition", "componentDefinition", klog.KObj(&componentDefinition))
 		return ctrl.Result{}, nil
 	}
 
-	// refresh package discover when componentDefinition is registered
-	if componentDefinition.Spec.Workload.Type != types.AutoDetectWorkloadDefinition {
-		err := utils.RefreshPackageDiscover(ctx, r.Client, r.dm, r.pd, &componentDefinition)
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/core/components/componentdefinition/componentdefinition_controller.go:86]: why RefreshPackageDiscover is unsafe?

Comment (barnettZQG) [reply pkg/controller/core.oam.dev/v1alpha2/core/components/componentdefinition/componentdefinition_controller.go:86]: It affects performance and the code has been commented a long time.

File: apis/core.oam.dev/v1beta1/policy_definition.go [hunk 2]
```diff
@@ -29,7 +29,8 @@ type PolicyDefinitionSpec struct {
 	// Reference to the CustomResourceDefinition that defines this trait kind.
 	Reference common.DefinitionReference `json:"definitionRef,omitempty"`
 
-	// Schematic defines the data format and template of the encapsulation of the policy definition
+	// Schematic defines the data format and template of the encapsulation of the policy definition.
+	// Only support the CUE category.
```
Comment (wonderflow) [inline apis/core.oam.dev/v1beta1/policy_definition.go:33]:
```suggestion
	// Only CUE schematic are supported for now.
```

File: apis/core.oam.dev/v1beta1/workflow_step_definition.go [hunk 3]
```diff
@@ -29,7 +29,8 @@ type WorkflowStepDefinitionSpec struct {
 	// Reference to the CustomResourceDefinition that defines this trait kind.
 	Reference common.DefinitionReference `json:"definitionRef,omitempty"`
 
-	// Schematic defines the data format and template of the encapsulation of the workflow step definition
+	// Schematic defines the data format and template of the encapsulation of the workflow step definition.
+	// Only support the CUE category.
```
Comment (wonderflow) [inline apis/core.oam.dev/v1beta1/workflow_step_definition.go:33]: same above

File: apis/core.oam.dev/v1beta1/core_types.go [hunk 4]
```diff
@@ -138,7 +138,8 @@ type TraitDefinitionSpec struct {
 	// +optional
 	ConflictsWith []string `json:"conflictsWith,omitempty"`
 
-	// Schematic defines the data format and template of the encapsulation of the trait
+	// Schematic defines the data format and template of the encapsulation of the trait.
+	// Only support the CUE and KUBE categories.
```
Comment (wonderflow) [inline apis/core.oam.dev/v1beta1/core_types.go:142]:
```suggestion
	// Only CUE and Kube schematic are supported for now.
```

**Review by wonderflow — APPROVED** [review]

**Pull request merged**